### PR TITLE
Feature 210 colored route

### LINKED ISSUE #210: Colored route
As a user I would like to have colored route on map

### PR DESCRIPTION
**Pull request by Chi-Chu319**
resolves #210

### CODE REVIEW

File: src/localization/fi.json [hunk 1]
```diff
@@ -261,6 +261,10 @@
     "userCreationError": "Virhe luotaessa käyttäjää.",
     "deleteRouteError": "Virhe poistettaessa reittiä.",
     "updateRouteList": "Virhe päivitettäessä reittiluetteloa.",
+    "listRoutingModes": "Error when list routing modes",
+    "listPollutant": "Error when listing pollutants",
+    "fetchRouteLine": "Error when fetching route line",
+    "listAirQuality": "Error when listing air quality",
```
Comment (Daniilmass) [inline src/localization/fi.json:267]: Should be in finnish
@Kaemppi Could you help?

File: src/utils/routing-utils.tsx [hunk 2]
```diff
@@ -121,4 +119,62 @@ export default class RoutingUtils {
...
+    const pollutantColorOpacity = Math.max(0, Math.min(fraction, 1));
+
+    const pollutantColor = lowColor.mix(highColor, pollutantColorOpacity);
+
+    const pollutantColorWithBase = _baseColor.mix(pollutantColor, baseColorOpacity);
+
+    return pollutantColorWithBase.hex();
+  }
+
+  public static stokeColorConverter = (pollutants: Pollutant[], routeAirQuality: AirQuality[], baseColor: string) =>{
```
Comment (Daniilmass) [inline src/utils/routing-utils.tsx:170]: Please add the TSdocs

File: src/utils/routing-utils.tsx [hunk 3]
```diff
@@ -121,4 +119,62 @@ export default class RoutingUtils {
...
+  /**
+   * Return a color for a pollutant
+   * 
+   * @param highDensityColor hight density color
+   * @param lowDensityColor low density color
+   * @param baseColor base color
+   * @param fraction fraction value
+   * @return pollutant color
+   */
+  public static getPollutantColorFaction = (baseColor: string, fraction: number): string => {
```
Comment (Daniilmass) [inline src/utils/routing-utils.tsx:156]: Please remove non-existing params from TSdocs

File: src/components/screens/main-screen.tsx [hunk 4]
```diff
@@ -702,17 +667,78 @@ class MainScreen extends React.Component<Props, State> {
...
 
   /**
-   * Returnes a route line represented by array of coordinates 
-   *
-   * @param defaultRoutingMode default routing mode data
-   * @returns array of coordinates
+   * Creates a route list offset
+   * 
+   * @param routeLine the route line as array of points
+   * @returns modified route line
```
Comment (Daniilmass) [inline src/components/screens/main-screen.tsx:673]:
```suggestion
   * Creates a route offset
   * 
   * @param routeAlts the route alternatives
   * @returns modified route lines
```
Should it be like that?

**Review by Daniilmass — CHANGES_REQUESTED** [review]

File: src/localization/fi.json (hunk 1 above)
Comment (Chi-Chu319) [reply src/localization/fi.json:267]: Totally missed this part...

Comment (Kaemppi) [inline src/localization/fi.json:267]:
```suggestion
    "listRoutingModes": "Virhe reittityyppien listaamisessa",
    "listPollutant": "Virhe saasteiden listaamisessa.",
    "fetchRouteLine": "Virhe reitin hakemisessa",
    "listAirQuality": "Virhe ilmanlaadun listauksessa",
```

**Pull request merged**